### PR TITLE
Support `Series` and Python primitives in `inplace_predict` and QDM

### PR DESCRIPTION
**Pull request by trivialfis**
- Support `pd.Series` in inplace predict.
- Support list and tuple in inplace predict.
- Support `pd.Series` in QDM.

### CODE REVIEW

File: tests/python/test_predict.py [hunk 1]
```diff
@@ -231,6 +231,7 @@ def test_base_margin(self):
         from_dmatrix = booster.predict(dtrain)
         np.testing.assert_allclose(from_dmatrix, from_inplace)
 
+    @pytest.mark.skipif(**tm.no_pandas())
```
Comment (hcho3) [inline tests/python/test_predict.py:234]: Is this necessary?

Comment (trivialfis) [reply tests/python/test_predict.py:234]: Yes. The test also uses pandas in `np_dtypes` for testing differences between np.dtype and pandas types. For instance, what happen if np.bool is passed in? Is it nullable in a pandas dataframe?

Comment (hcho3) [reply tests/python/test_predict.py:234]: Still not sure. Neither `np_dtypes` nor `test_dtypes` has any reference to Pandas?

Comment (trivialfis) [reply tests/python/test_predict.py:234]: https://github.com/dmlc/xgboost/blob/0c38ca7f6e329dbb062939dcb843d0dc229675b4/python-package/xgboost/testing/data.py#L46

**Review by hcho3 — APPROVED** [review]

**Pull request merged**